src/daft-functions-list/src/list_map.rs

```diff
@@ -0,0 +1,47 @@
+use daft_core::{array::ListArray, series::IntoSeries};
+use daft_dsl::functions::prelude::*;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ListMap;
+
+#[derive(FunctionArgs)]
+struct ListMapArgs<T> {
+    input: T,
+    expr: T,
+}
+
+#[typetag::serde]
+impl ScalarUDF for ListMap {
```

Comment:
I would add some comments to this impl to describe what we're expecting as the inputs before and after resolution. 

Before: `expr` should have element expressions as placeholders
After: those placeholders should be replaced with the `input`, exploded